### PR TITLE
samples: caf: Add nrf54h20pdk configuration

### PR DESCRIPTION
**Pull request by pdunaj**
Add configuration for nRF54H20PDK.

### CODE REVIEW

**Review by rakons — APPROVED** [review]

**Comment by jfischer-no** [comment]
@pdunaj please get some more +1

File: samples/caf/boards/nrf54h20pdk_nrf54h20_cpuapp.overlay [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+ /{
+	/delete-node/ leds;
+
+	leds0 {
+		compatible = "gpio-leds";
+		led0: led_0 {
+			gpios = < &gpio9 0 GPIO_ACTIVE_HIGH >;
```
Comment (gmarull) [inline samples/caf/boards/nrf54h20pdk_nrf54h20_cpuapp.overlay:13]:
```suggestion
			gpios = <&gpio9 0 GPIO_ACTIVE_HIGH>;
```

in all places

File: samples/caf/boards/nrf54h20pdk_nrf54h20_cpuapp.overlay [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+		};
+	};
+
+	leds3 {
+		compatible = "gpio-leds";
+		led3: led_3 {
+			gpios = < &gpio9 3 GPIO_ACTIVE_HIGH >;
+			label = "Green LED 3";
+		};
+	};
```
Comment (gmarull) [inline samples/caf/boards/nrf54h20pdk_nrf54h20_cpuapp.overlay:40]: question: why is this repeating information already upstream? https://github.com/zephyrproject-rtos/zephyr/blob/main/boards/nordic/nrf54h20pdk/nrf54h20pdk_nrf54h20_cpuapp.dts

There's no need to have one node per led, `gpio-leds` doesn't work like that for cases like this.

**Review by gmarull — CHANGES_REQUESTED** [review]

File: samples/caf/boards/nrf54h20pdk_nrf54h20_cpuapp.overlay (hunk 1 above)
Comment (pdunaj) [reply samples/caf/boards/nrf54h20pdk_nrf54h20_cpuapp.overlay:13]: will do

File: samples/caf/boards/nrf54h20pdk_nrf54h20_cpuapp.overlay (hunk 2 above)
Comment (pdunaj) [reply samples/caf/boards/nrf54h20pdk_nrf54h20_cpuapp.overlay:40]: Our application framework use dts leds grouping to handle physical LEDs into logical (RGB or monochromatic) LEDs.
Upstream does not match what we want.

Comment (gmarull) [reply samples/caf/boards/nrf54h20pdk_nrf54h20_cpuapp.overlay:40]: Then this needs to be fixed upstream, it makes no sense to keep copies of stuff like this downstream.

**Review by KAGA164 — APPROVED** [review]

File: samples/caf/boards/nrf54h20pdk_nrf54h20_cpuapp.overlay (hunk 2 above)
Comment (pdunaj) [reply samples/caf/boards/nrf54h20pdk_nrf54h20_cpuapp.overlay:40]: This is application specific but to be honest I don't mind if DKs follow the configuration that we set up.

**Comment by pdunaj** [comment]
the first force push is a rebase, the second one is requested formatting fix

**Review by shanthanordic — APPROVED** [review]

**Pull request merged**